pulsar-broker/src/main/java/org/apache/pulsar/broker/transaction/buffer/impl/TopicTransactionBuffer.java

```diff
@@ -197,7 +173,8 @@ public void handleTxnEntry(Entry entry) {
                             PositionImpl position = PositionImpl.get(entry.getLedgerId(), entry.getEntryId());
                             if (Markers.isTxnMarker(msgMetadata)) {
                                 if (Markers.isTxnAbortMarker(msgMetadata)) {
-                                    aborts.put(txnID, position);
+                                    snapshotAbortedTxnProcessor.appendAbortedTxn(
+                                            new TxnIDData(txnID.getMostSigBits(), txnID.getLeastSigBits()), position);
```

Comment:
```suggestion
                                    snapshotAbortedTxnProcessor.appendAbortedTxn(txnId, position);
```
TxnIDData only take snapshot need to cover